### PR TITLE
Dpg code gen fix docs

### PR DESCRIPTION
**Pull request by joheredi**
The classical client was missing method names. Adding them, also a couple of other fixes:

- Remove calculated type for deserialize operation parameter
- Added return type to send function
- renamed client options after the client

### CODE REVIEW

**Comment by MaryGao** [comment]
may fix this https://github.com/Azure/autorest.typescript/issues/1841

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 1]
```diff
@@ -15,6 +15,17 @@ import { buildType } from "./typeHelpers.js";
 import { getOperationName } from "./namingHelpers.js";
 import { getFixmeForMultilineDocs } from "./fixmeHelpers.js";
 
+function getRLCResponseType(operation: Operation) {
+  const { name } = getOperationName(operation, { casing: "pascal" });
+  const statusCodes: (string | number)[] = Array.from(
+    new Set(operation.responses.flatMap((r) => r.statusCodes)).values()
+  );
+
+  statusCodes.push("Default");
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:24]: Should we consider LogicalResponse here ? @MaryGao

File: packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/EventGridClient.ts [hunk 2]
```diff
@@ -25,18 +25,21 @@ import {
...
+
 export class EventGridClient {
   private _client: EventGridContext;
 
   /** Azure Messaging EventGrid Client */
   constructor(
     endpoint: string,
     credential: KeyCredential,
-    options: ClientOptions = {}
+    options: EventGridClientOptions = {}
```
Comment (qiaozha) [inline packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/EventGridClient.ts:37]: Just a reminder, may need to check the ClientOptions in the rest level. Also, should we generate EventGridClientOptions if there's no actual difference between EventGridClientOptions and ClientOptions ? For RLC, the answer is no, right now.

**Review by qiaozha — APPROVED** [review]

File: packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/EventGridClient.ts (hunk 2 above)
Comment (joheredi) [reply packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/EventGridClient.ts:37]: Let's add the logic to remove if no client level parameters in a separate PR

**Pull request merged**